JumpscaleLibs/sal/zosv2/Zosv2.py

```diff
@@ -0,0 +1,197 @@
+        wg_port = self.find_free_wg_port(node)
+
+        nr1 = network.network_resources.new()
+
+        nr1.iprange = network1
+        nr1.node_id = node.node_id
+        nr1.wireguard_listen_port = wg_port
+        nr1.wireguard_public_key = wg_public
+        nr1.wireguard_private_key_encrypted = wg_private_encrypted
+
+        # add 256 to plus 1 in network from 172.20.0.0 -> 172.20.1.0
+        network_range += 256
+        network2 = str(network_range) + "/24"
+        ip = netaddr.IPAddress(network_range)
+        allowed_ip = ip.words
+
+        private_key, public_key = self.key_pair_get()
+        # add a peer to the network, this peer is your laptop
+        peer = nr1.peers.new()
+        peer.iprange = network2
+
+        peer.allowed_iprange = ["100.64." + str(allowed_ip[1]) + "." + str(allowed_ip[2]) + "/32", network2]
+        # this is your wireguard public key from your laptop
+        peer.public_key = public_key.decode()
+
+        self.config.public_key = public_key
+        self.config.private_key = private_key
+        self.config.wg_port = wg_port
+
+        return network, self.config
+
+    def add_volume(self,reservation, node_id, voulme_size=5, volume_type="SSD"):
+        volume = reservation.data_reservation.volumes.new()
+        volume.workload_id = 2
+        volume.size = voulme_size
+        volume.type = volume_type
+        volume.node_id = node_id
+        return  volume
+
+    def get_all_ips(self, ip_range):
+        networks = netaddr.IPNetwork(ip_range)
+        ips = []
+        for ip in list(networks.iter_hosts()):
+           ips.append(ip.format())
+        return ips
+
+    def create_container(self,network_name,ip_address,node_id, reservation, flist, env={}, entrypoint="",interactive=False, volume_size=0, voulme_type="SSD", mount_point="/sandbox/var"):
+
+        cont = reservation.data_reservation.containers.new()
+        cont.node_id = node_id
+        cont.workload_id = 3
+
+        cont.flist = flist
+        cont.storage_url = "zdb://hub.grid.tf:9900"
+        cont.environment = env
+        cont.entrypoint = entrypoint
+        cont.interactive = interactive
+
+        net = cont.network_connection.new()
+        net.network_id = network_name
+        net.ipaddress = ip_address
+
+        if volume_size > 0:
+            volume = self.add_volume(reservation,node_id,volume_size,voulme_type)
+            vol = cont.volumes.new()
+            # here we reference the volume created in the same reservation
+            vol.workload_id = 3
+            vol.volume_id = f"-{volume.workload_id}"
+            vol.mountpoint = mount_point
+
+        return reservation
+
+
+    def wg_config(self, ip_range,network_config):
+
+        first_network_range = netaddr.IPNetwork(ip_range).ip
+        ip = netaddr.IPAddress(first_network_range)
+        first_ip = ip.words
+        # add 256 + 256 to 172.20.0.0 -> 172.20.2.0
```

Comment:
here again this weird transformation on the ip_range for no reason...

Reply:
i'm not sure , but i did it to get configs of Wire-guard
example from :
https://github.com/threefoldtech/jumpscaleX_threebot/blob/development/ThreeBotPackages/tfgrid/provision_examples/ubuntu_ssh.py#L56

Reply:
After check this md file https://github.com/threefoldtech/zos/blob/master/docs/network/mesh.md#interworkings
that mean parent network is provided by the user and for each node added to this network it has to be assigned to a subnet of the parent network
parent network -> 172.20.0.0/16 ( chosen by user)
Node -> 172.20.1.0/24
host -> 172.20.2.0/24